packageurl.go

```diff
@@ -380,6 +380,10 @@ func pathEscape(s string) string {
 		switch {
 		case c == '@':
 			t.WriteString("%40")
+		case c == '+':
+			// url.PathEscape doesn't encode '+' since it's a valid query escape character for ' ' in application/x-www-form-urlencoded, but '+' is a
+			// valid character in semver so we don't want it to be unintentionally unescaped as ' ' by downstream parsers of the purl.
+			t.WriteString("%2B")
```

Comment:
note that `url.QueryEscape` will encode `+`, but that method was replace with `PathEscape` in https://github.com/anchore/packageurl-go/pull/1